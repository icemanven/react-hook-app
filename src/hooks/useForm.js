import { useState } from "react";

export const useForm = (initialState = {}) => {
  const [values, setValues] = useState(initialState);

  const hadleInputChange = ({target}) => {
      setValues({
          ...values,
          [target.name]: target.value
      });
  };

  const resetForm = (iniData = initialState) => {
    setValues(iniData);
  }

  return [values, hadleInputChange, resetForm];

}
