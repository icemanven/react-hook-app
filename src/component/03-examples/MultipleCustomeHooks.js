import React from "react";
import { useCounter } from "../../hooks/useCounter";
import { useFetch } from "../../hooks/useFetch";

import "../02-useEffect/effects.css";

export const MultipleCustomeHooks = () => {
  const { counter, increment } = useCounter(1);

  const { data, loading } = useFetch(
    `https://www.breakingbadapi.com/api/quotes/${counter}`
  );

  const {quote, author} = !!data && data[0];

  return (
    <div>
      <h1>Breaking bad Quoates</h1>
      <hr />

      {loading ? (
        <div className="alert alert-info text-center">loading...</div>
      ) : (
        <blockquote className=" text-right blockquote">
          <p className="mb-10">{quote}</p>
          <footer className="blockquote-footer">{author}</footer>
        </blockquote>
      )}

      {!loading && (
        <button onClick={() => increment()} className="btn btn-primary">
          Siguiente frase
        </button>
      )}
    </div>
  );
};
