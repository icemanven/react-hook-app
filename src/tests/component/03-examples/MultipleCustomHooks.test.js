import React from "react";
import { shallow } from "enzyme";
import { MultipleCustomeHooks } from "../../../component/03-examples/MultipleCustomeHooks";
import { useFetch } from "../../../hooks/useFetch";
import { useCounter } from "../../../hooks/useCounter";

jest.mock("../../../hooks/useFetch");
jest.mock("../../../hooks/useCounter");

describe("Pruebas en <MultipleCustomeHooks />", () => {


  test("debe mostrarse correctamente", () => {
    useFetch.mockReturnValue({
      data: null,
      loading: true,
      error: null,
    });

    useCounter.mockReturnValue({
        counter: 10,
        increment: () => {},
        decrement: () => {},
        reset: () => {},
      });
    
    const wrapper = shallow(<MultipleCustomeHooks />);

    expect(wrapper).toMatchSnapshot();
  });

  test("debe mostrar la informacion", () => {
    useFetch.mockReturnValue({
      data: [
        {
          author: "fernando",
          quote: "Hola mundo",
        },
      ],
      loading: false,
      error: null,
    });

    useCounter.mockReturnValue({
        counter: 10,
        increment: () => {},
        decrement: () => {},
        reset: () => {},
      });

    const wrapper = shallow(<MultipleCustomeHooks />);

    expect(wrapper.find(".alert").exists()).toBe(false);
    expect(wrapper.find(".mb-10").text().trim()).toBe("Hola mundo");
    expect(wrapper.find("footer").text().trim()).toBe("fernando");
  });
});
