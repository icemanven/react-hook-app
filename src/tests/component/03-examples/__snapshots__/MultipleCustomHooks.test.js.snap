// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <MultipleCustomeHooks /> debe mostrarse correctamente 1`] = `
<div>
  <h1>
    Breaking bad Quoates
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    loading...
  </div>
</div>
`;
